bin/faucet/src/server/get_tokens.rs

```diff
@@ -164,7 +167,12 @@ impl RawMintRequest {
         )?;
 
         // Check the PoW solution
-        check_pow_solution(&self.pow_seed, self.pow_solution)?;
+        check_pow_solution(
+            challenge_state,
+            &self.pow_seed,
+            &self.server_signature,
+            self.pow_solution,
+        )?;
```

Comment:
nit: Because this takes various arguments from `self`, would it make sense to make it a member function of `RawMintRequest`?

Reply:
Also, should checking that the challenge exists in the cache come before the other checks? So that if it does not exist we can exit early (could be done here or even before calling `validate()`).